src/App.tsx

```diff
@@ -1,204 +1,62 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { TodoFooter } from './components/footer/TodoFooter';
+import { Errors } from './components/Errors/Errors';
+import { TodoRenderList } from './components/main/TodoRenderList';
+import { TodoHeader } from './components/header/Header';
+import { Todo } from './types/Todo';
+import { getTodos } from './api/todos';
+import { filterTodos } from './helpers/helper';
+import { TypeOfFilter } from './types/TypeOfFilters';
+import { Error } from './types/TypeOfErrors';
+
+const USER_ID = 12049;
 
 export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState<TypeOfFilter>(TypeOfFilter.All);
+  const [error, setError] = useState<Error | null>(null);
+
+  const filteredTodos = useMemo(() => {
+    return filterTodos(todos, filter);
+  }, [todos, filter]);
+
+  useEffect(() => {
+    setError(null);
```

Comment:
error has `null` as initial value, no need to set it again here, because useEffect has an empty array of deps so component will have state `error` `null` before mounting